easybuild/easyblocks/g/gcc.py

```diff
@@ -775,9 +879,25 @@ def run_all_steps(self, *args, **kwargs):
         iteration 1 builds nvptx-tools
         iteration 2 builds the nvptx target compiler
         """
+        self.cfg['configopts'] = ['']
+        self.cfg['buildopts'] = ['']
+        stage = 1
         if self.cfg['withnvptx']:
-            self.cfg['configopts'] = [self.cfg['configopts']] * 3
-            self.cfg['buildopts'] = [self.cfg['buildopts']] * 3
+            # Add two additional configure and build stages when enabling NVPTX
+            # this is required to first build NVPTX tools and then newlib
+            self.cfg['configopts'] += ['', '']
+            self.cfg['buildopts'] += ['', '']
+            self.build_stages[stage] = NVPTX_TOOLS
+            self.build_stages[stage + 1] = NVIDIA_NEWLIB
+            stage += 2
+        if self.cfg['withamdgcn']:
+            # Add two additional stages when enabling AMD GCN
+            # this is required to first build LLVM tools and then newlib
+            self.cfg['configopts'] += ['', '']
+            self.cfg['buildopts'] += ['', '']
```

Comment:
Avoid duplicating this:

```python
if self.cfg['withnvptx'] or self.cfg['withamdgcn']:
    # sensible comment goes here
    self.cfg['configopts'] += ['', '']
    self.cfg['buildopts'] += ['', '']
    stage += 2
    if self.cfg['withnvptx']:
         self.build_stages[1] = NVPTX_TOOLS
         self.build_stages[2] = NVIDIA_NEWLIB
    elif self.cfg['withamdgcn']:
         self.build_stages[1] = NVPTX_TOOLS
         self.build_stages[2] = NVIDIA_NEWLIB
    else:
         # error out?

Reply:
It is possible to build both during the same install, that is the reason for the duplication and `stage` variable